monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,379 @@
+
+  def testunit_assert_collectd_installed_WHEN_bad_socket_RAISES_CriticalError(
+      self):
+    check_collectd_mlab.COLLECTD_BIN = (
+        os.path.join(self._testdata_dir, 'fake_bin'))
+    check_collectd_mlab.COLLECTD_NAGIOS = (
+        os.path.join(self._testdata_dir, 'fake_bin'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingSocketCriticalError,
+        check_collectd_mlab.assert_collectd_installed)
+
+  @mock.patch('os.access')
+  def testunit_assert_collectd_responds_WHEN_filesystem_is_readonly(
+      self, mock_access):
+    # Mock out os.access to guarantee that write access is rejected.
+    mock_access.return_value = False
+
+    self.assertRaises(
+        check_collectd_mlab.ReadonlyFilesystemCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+    self.assertTrue(mock_access.called)
+
+  def testunit_assert_collectd_responds_WHEN_sock_connect_fails(self):
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.SocketConnectionCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_backend_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysBackendCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_slice_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysFrontendCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_vsys_acl_is_missing(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = os.path.join(
+        self._testdata_dir, 'fake_slice')
+    check_collectd_mlab.VSYSPATH_ACL = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingVsysAclCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
+
+  def testunit_assert_collectd_vsys_setup_WHEN_acl_incomplete(self):
+    check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(
+        self._testdata_dir, 'fake_backend')
+    check_collectd_mlab.VSYSPATH_SLICE = os.path.join(
+        self._testdata_dir, 'fake_slice')
+    check_collectd_mlab.VSYSPATH_ACL = os.path.join(
+        self._testdata_dir, 'fake_acl')
+
+    self.assertRaises(
+        check_collectd_mlab.MissingSliceFromVsysAclCriticalError,
+        check_collectd_mlab.assert_collectd_vsys_setup)
```

Comment:
This is a bit hard to understand. Is the test supposed to fail because the ACL doesn't include the slice name? It's confusing because `fake_backend` is supposed to be good, `fake_slice` is supposed to be good, so it's unintuitive that `fake_acl` is bad. Consider naming it something like `acl_missing_slice_name`.